### PR TITLE
Remove old migrations!

### LINKED ISSUE #5215: Archive/delete old migrations
It would be really nice if we could delete/archive our old migrations. We have **442** migrations going back to **2012** and every one of them is run during every deploy. 

This makes the database migration step really slow, and is basically completely pointless, as all those state changes have already been applied. 

Having a huge pile of ancient migrations also increases the chance that at some point we'll change or delete some models in a way that breaks things.

I think we can do something (roughly) like this:
https://medium.com/clutter-engineering/cleaning-up-old-rails-migrations-1b55b638abb5

### PR DESCRIPTION
**Pull request by Matt-Yorkley**
#### What? Why?

Closes #5215 

Deletes all migrations prior to 2019. That's ~6 years worth of migrations we don't need to touch anymore! :tada:

Follows the tip from [this article](https://medium.com/clutter-engineering/cleaning-up-old-rails-migrations-1b55b638abb5) which adds a nice note for any old developers that are returning to OFN but haven't loaded their environment since 2018, to explain why `db:migrate` isn't working and how to fix it. :100:

I considered moving the migrations to a separate folder like `/archived_migrations` for reference, but if anyone really wants to go on an archeological expedition into our database state changes from 2012 they can look in the git history. #YAGNI :fire:

#### What should we test?

- Running `bundle exec rake db:migrate` works.
- Check out an old schema and make sure that the error appears correctly:
  ```
  git checkout 9840467b8af9fae3a1785dc64ae75865bee6816a -- db/schema.rb
  bundle exec rake db:reset
  bundle exec rake db:migrate
  ```
- Deploying works

#### Release notes

Deleted all legacy database migration files prior to 2019.

Changelog Category: Removed

### CODE REVIEW

**Comment by Matt-Yorkley** [comment]
Screenshot from 2020-04-16 14-00-30

 :smile:

**Comment by mkllnk** [comment]
After my changes, the error looks like this:
```
==  OldMigrationsRemoved: migrating ===========================================
rake aborted!
StandardError: An error has occurred, this and all later migrations canceled:


      You haven't updated your dev environment in a long time!
      Legacy migration files before 2019 have now been removed.
      Run `rake db:schema:load` before running `rake db:migrate`.

/home/maikel/code/openfoodnetwork/db/migrate/20181128054803_old_migrations_removed.rb:5:in `up'
/home/maikel/.rbenv/versions/2.3.7/bin/bundle:22:in `load'
/home/maikel/.rbenv/versions/2.3.7/bin/bundle:22:in `<main>'

Caused by:
StandardError: 
      You haven't updated your dev environment in a long time!
      Legacy migration files before 2019 have now been removed.
      Run `rake db:schema:load` before running `rake db:migrate`.

/home/maikel/code/openfoodnetwork/db/migrate/20181128054803_old_migrations_removed.rb:5:in `up'
/home/maikel/.rbenv/versions/2.3.7/bin/bundle:22:in `load'
/home/maikel/.rbenv/versions/2.3.7/bin/bundle:22:in `<main>'
Tasks: TOP => db:migrate
(See full trace by running task with --trace)
```

**Comment by mkllnk** [comment]
I staged this and it seemed to work fine. I'm not sure if someone else should test this again though because I'm one of the authors now.

**Comment by luisramos0** [comment]
nice Maikel, lets have @sauloperez have a look as well, and maybe he can merge this directly

File: db/migrate/20181128054803_old_migrations_removed.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class OldMigrationsRemoved < ActiveRecord::Migration
+  LAST_DELETED_MIGRATION = 20181128054803
+  def up
+    if ActiveRecord::Migrator.current_version < LAST_DELETED_MIGRATION
```
[the remark this answers is not in the document]
Comment (luisramos0) [reply db/migrate/20181128054803_old_migrations_removed.rb]: I think you are right Maikel.

**Pull request merged**